app/screens/channel/channel.android.js

```diff
@@ -29,6 +31,29 @@ export default class ChannelAndroid extends ChannelBase {
         openSettingsSideMenu();
     };
 
+    componentDidMount() {
+        BackHandler.addEventListener('hardwareBackPress', this.handleBackPress.bind(this));
+    }
+
+    handleBackPress() {
+        if (backPressedCount > 0) {
+            BackHandler.exitApp();
+            backPressedCount = 0;
+        } else {
+            backPressedCount++;
+            ToastAndroid.show('Press Again To Exit', ToastAndroid.SHORT);
+            setTimeout(() => {
+                backPressedCount = 0;
+            }, 2000);
+            return true;
+        }
+        return false;
+    }
```

Comment:
```suggestion
    handleBackPress = () => {
        if (!backPressedCount && EphemeralStore.getNavigationTopComponentId() === 'Channel') {
            const {formatMessage} = this.context.intl;

            backPressedCount++;
            ToastAndroid.show(formatMessage({
                id: 'mobile.android.back_handler_exit',
                defaultMessage: 'Press again to exit',
            }), ToastAndroid.SHORT);

            setTimeout(() => {
                backPressedCount = 0;
            }, 2000);

            return true;
        }

        backPressedCount = 0;
        return false;
    };
```

Reasons for this change:
- Because this component is mounted at all times, you need to confirm that the top navigation component is actually the `Channel` screen, not an open drawer or any of the other screens. for this you'll need to import EphemeralStorage `import EphemeralStore from '@store/ephemeral_store';`
- You don't need to call `BackHandler.exitApp()` as that would  be the default behavior when this handler returns false.
- The text message needs to be translatable

Once the translatable string is added, you'll need to execute `make i18n-extract` so it gets added to the english translation file, then once the PR gets merge it will show up for translation in the translation server.

Reply:
@enahum I have fixed the feedbacks. Thanks for the detailed feedback. Learnt a lot from this:)